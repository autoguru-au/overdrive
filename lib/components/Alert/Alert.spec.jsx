import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Alert } from '.';

describe('<Alert />', () => {
	it('should not throw', () => {
		expect(() => render(<Alert />)).not.toThrow();
	});

	it('should match the snapshot', () => {
		expect(
			render(<Alert onRequestClose={jest.fn()} />).container.firstChild,
		).toMatchSnapshot();
	});

	it('should fire callback when X is clicked', () => {
		const closeHandler = jest.fn();

		const { getByLabelText } = render(
			<Alert onRequestClose={closeHandler} />,
		);

		fireEvent.click(getByLabelText('close'));

		expect(closeHandler).toHaveBeenCalledTimes(1);
	});
});
