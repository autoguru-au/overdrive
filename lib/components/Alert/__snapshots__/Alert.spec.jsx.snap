// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match the snapshot 1`] = `
<div
  class="root info dismissible"
  role="alert"
>
  <i
    class="root icon"
    style="width: 20px; height: 20px;"
  >
    <svg
      baseProfile="full"
      fill="currentColor"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 8.998h-2v-2h2m0 10h-2v-6h2m-1-9c-5.524 0-10 4.477-10 10s4.476 10 10 10c5.522 0 10-4.477 10-10s-4.478-10-10-10z"
      />
    </svg>
  </i>
  <div
    class="content"
  />
  <button
    aria-label="close"
    class="root variantSecondary small minimal iconOnly closeButton"
    type="button"
  >
    <div
      class="body"
    >
      <i
        class="root closeButtonIcon"
        style="width: 20px; height: 20px;"
      >
        <svg
          baseProfile="full"
          fill="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M13.458 12L19 17.543V19h-1.457L12 13.457 6.458 19H5v-1.456L10.544 12 5 6.458V5h1.457L12 10.543 17.544 5H19v1.458L13.458 12z"
          />
        </svg>
      </i>
    </div>
  </button>
</div>
`;
